src/include/module/module/interface.h

```diff
@@ -0,0 +1,222 @@
+	 *	    - sources[].data is a pointer to raw audio data
+	 *	- sinks are output_stream_buffer[]
+	 *	    - sinks[].data is a pointer to raw audio data
+	 */
+	int (*process_raw_data)(struct processing_module *mod,
+				struct input_stream_buffer *input_buffers,
+				int num_input_buffers,
+				struct output_stream_buffer *output_buffers,
+				int num_output_buffers);
+
+	/**
+	 * Set module configuration for the given configuration ID
+	 *
+	 * If the complete configuration message is greater than MAX_BLOB_SIZE bytes, the
+	 * transmission will be split into several smaller fragments.
+	 * In this case the ADSP System will perform multiple calls to SetConfiguration() until
+	 * completion of the configuration message sending.
+	 * \note config_id indicates ID of the configuration message only on the first fragment
+	 * sending, otherwise it is set to 0.
+	 */
+	int (*set_configuration)(struct processing_module *mod,
+				 uint32_t config_id,
+				 enum module_cfg_fragment_position pos, uint32_t data_offset_size,
+				 const uint8_t *fragment, size_t fragment_size, uint8_t *response,
+				 size_t response_size);
+
+	/**
+	 * Get module runtime configuration for the given configuration ID
+	 *
+	 * If the complete configuration message is greater than MAX_BLOB_SIZE bytes, the
+	 * transmission will be split into several smaller fragments.
+	 * In this case the ADSP System will perform multiple calls to GetConfiguration() until
+	 * completion of the configuration message retrieval.
+	 * \note config_id indicates ID of the configuration message only on the first fragment
+	 * retrieval, otherwise it is set to 0.
+	 */
+	int (*get_configuration)(struct processing_module *mod,
+				 uint32_t config_id, uint32_t *data_offset_size,
+				 uint8_t *fragment, size_t fragment_size);
+
+	/**
+	 * Set processing mode for the module
+	 */
+	int (*set_processing_mode)(struct processing_module *mod,
+				   enum module_processing_mode mode);
+
+	/**
+	 * Get the current processing mode for the module
+	 */
+	enum module_processing_mode (*get_processing_mode)(struct processing_module *mod);
+
+	/**
+	 * Module specific reset procedure, called as part of module_adapter component
+	 * reset in .reset(). This should reset all parameters to their initial stage
+	 * and free all memory allocated during prepare().
+	 */
+	int (*reset)(struct processing_module *mod);
+	/**
+	 * Module specific free procedure, called as part of module_adapter component
+	 * free in .free(). This should free all memory allocated during module initialization.
+	 */
+	int (*free)(struct processing_module *mod);
+	/**
+	 * Module specific bind procedure, called when modules are bound with each other
+	 */
+	int (*bind)(struct processing_module *mod, void *data);
+	/**
+	 * Module specific unbind procedure, called when modules are disconnected from one another
+	 */
+	int (*unbind)(struct processing_module *mod, void *data);
+
+#ifdef SOF_MODULE_API_PRIVATE
```

Comment:
Not a blocker, but we may want to add some commentry here to explain what private means.